src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -97,7 +97,7 @@ SQLStorage::SQLStorage(const StorageConfig& config) : INvStorage(config) {
 }
 
 void SQLStorage::storePrimaryKeys(const std::string& public_key, const std::string& private_key) {
-  SQLite3Guard db(dbPath());
+  SQLite3Guard db(dbPath(), readonly_);
```

Comment:
Maybe it's time to set up a private helper method, the boiler-plate is starting to get a bit heavy:

```
SQLite3Guard SQLStorage::dbConnection() {
  SQLite3Guard db(dbPath(), readonly_);
  if (db.get_rc() != SQLITE_OK) {
    LOG_ERROR << "Can't open database: " << db.errmsg();
  }
  return std::move(db);
}
```

And usage:

```
  SQLite3Guard db = dbConnection();
  if (db.get_rc() != SQLITE_OK) {
    return false;
  }
```

Reply:
Nice idea